### PR TITLE
refactor: Refactor TKs

### PR DESCRIPTION
**Pull request by reginawang3495**
## Summary
* Refactor all []TypeAndKey declarations to TKs
* Rename TypeAndKey type to TK
## Test Plan
`cd $MAGMA_ROOT/orc8r/cloud/docker && ./build.py --tests ; noti`

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

**Comment by andreilee** [comment]
Thanks for doing this refactor!
One thing to mention:
You'll need to sign off your commits - try `git commit -s --amend` to add a sign off

**Comment by reginawang3495** [comment]
@andreilee Thanks for the info! I think I've signed off now :0

~~Are the failing checks ok? They pass successfully on my computer. Also, should I wait for all checks to run before tagging reviewers?~~

EDIT: Hunter answered this for me :)

**Review by hcgatewood — APPROVED** [review]
Skimmed, looks great, thanks for the refactor! Msg me when you're ready to merge

**Review by themarwhal — APPROVED** [review]

File: src/go/go.mod [hunk 1]
```diff
@@ -22,19 +22,3 @@ require (
 	google.golang.org/grpc/cmd/protoc-gen-go-grpc v1.1.0
 	google.golang.org/protobuf v1.27.1
 )
-
```
Comment (themarwhal) [inline src/go/go.mod]: Actually @reginawang3495 can you revert this change? @markjen / @hcgatewood  I'm not sure why this keeps happening, but some unrelated Golang development seems to affect this file.

File: magma.iml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module version="4">
+  <component name="Go" enabled="true" />
+  <component name="NewModuleRootManager" inherit-compiler-output="true">
+    <exclude-output />
+    <content url="file://$MODULE_DIR$" />
+    <orderEntry type="sourceFolder" forTests="false" />
+  </component>
+</module>
```
Comment (reginawang3495) [inline magma.iml]: I've manually deleted this for now, but I think this gets created when adding a module. Should `*.iml` files be `.gitignore`d?

> An IML file is a module settings file created by IntelliJ IDEA, an Integrated Development Environment (IDE) used to develop Java applications. It stores information about a development module, which is a Java, Plugin, Android, or Maven component of a Java application. IML files store a module’s type, paths, dependencies, and other settings.

[Source](https://fileinfo.com/extension/iml)

File: src/go/go.mod (hunk 1 above)
Comment (reginawang3495) [reply src/go/go.mod]: I think when I delete the changes here + `magma.iml`, I need to import a new magma module 
image

When I create a new magma module, these 2 files get modified (sometimes, I believe `~/magma/go.mod` also changes)

Comment (reginawang3495) [reply src/go/go.mod]: I think when I import `magma` as a new module, this part gets deleted + `go.mod` 
```
module magma

go 1.13
```
gets created

Comment (reginawang3495) [reply src/go/go.mod]: Hunter said I can leave for now

File: magma.iml (hunk 2 above)
Comment (reginawang3495) [reply magma.iml]: Ye will add to .gitignore
